api/v1alpha4/azuremachinetemplate_webhook.go

```diff
@@ -30,3 +39,36 @@ func (r *AzureMachineTemplate) SetupWebhookWithManager(mgr ctrl.Manager) error {
 		For(r).
 		Complete()
 }
+
+// +kubebuilder:webhook:verbs=update,path=/validate-infrastructure-cluster-x-k8s-io-v1alpha4-azuremachinetemplate,mutating=false,failurePolicy=fail,matchPolicy=Equivalent,groups=infrastructure.cluster.x-k8s.io,resources=azuremachinetemplates,versions=v1alpha4,name=validation.azuremachinetemplate.infrastructure.cluster.x-k8s.io,sideEffects=None,admissionReviewVersions=v1beta1
+
+var _ webhook.Validator = &AzureMachineTemplate{}
+
+// ValidateCreate implements webhook.Validator so a webhook will be registered for the type
+func (r *AzureMachineTemplate) ValidateCreate() error {
+	return nil
+}
+
+// ValidateUpdate implements webhook.Validator so a webhook will be registered for the type
+func (r *AzureMachineTemplate) ValidateUpdate(oldRaw runtime.Object) error {
+	clusterlog.Info("validate update", "name", r.Name)
+	var allErrs field.ErrorList
+	old := oldRaw.(*AzureMachineTemplate)
+
+	if !reflect.DeepEqual(r, old) {
```

Comment:
This works but the error message won't be very intuitive. Should we check each field separately, and return errors for invalid fields?
cc @CecileRobertMichon @nader-ziada

Reply:
+1.  But ```AzureMachineTemplate ``` has many fields , check for each field would be a little complicated...